### PR TITLE
Cross-book hyperlink support

### PR DESCRIPTION
**Pull request by fabienrohrer**
Attempt to fix: https://github.com/omichel/webots-doc/issues/327
Preview: https://www.cyberbotics.com/doc/guide/computer-peripherals?version=enhancement-cross-book-hyperlinks

- [x] Create a cross-book hyperlink system
- [x] Test it
    - [x] On production website
    - [x] Locally
    - [x] Embedded in Webots
    - [x] In GitHub Markdown Viewer
- [x] Fix existing links accordingly
- [x] Test that cross-book hyperlinks are consistent

### CODE REVIEW

**Comment by fabienrohrer** [comment]
I think the following MD notation is the best, because it is compliant (and working) in the Github Markdown Viewer, and it is consistent with the images urls:

``` Markdown
[url name](../BOOK/PAGE.md)
```

**Comment by fabienrohrer** [comment]
`tests/test_references.py` is already compilant with this case. ❤️

**Comment by fabienrohrer** [comment]
👍

File: guide/how-to.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ Then, the motor force is turned off and the boxes oscillate for a while, accordi
 In the first layout, the four motorized wheels are positioned circularly, as could be done for omnidirectional-wheeled robots.
 In the second layout, the four motorized wheels are oriented in the same direction, as could be done for a robot with tracks.
 In the third layout, a simple [Ackermann steering geometry](https://en.wikipedia.org/wiki/Ackermann_steering_geometry) is shown.
-Note that more completed Ackermann steering geometry can be achieved using the `AckermannVehicle.proto` model, and the `vehicle` library (cf. the [automobile book](https://www.cyberbotics.com/doc/automobile/index)).
+Note that more completed Ackermann steering geometry can be achieved using the [`AckermannVehicle` PROTO](../automobile/ackermannvehicle.md), and the [`vehicle` library](../automobile/driver-library.md).
```
Comment (omichel) [inline guide/how-to.md]: It seems wrong: it should be "... and the [`car` library](../automobile/car-library.md).". Isn't it?

File: guide/programming.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ You will find more info about *odometry* and *SLAM* techniques in `Cyberbotics'
 
 To get the position of a robot in the robot's controller code: add a `GPS` node to the robot, then use the `wb_robot_get_device`, `wb_gps_enable` and `wb_gps_get_values` functions.
 Note that the `GPS`'s resolution field must be 0 (the default), otherwise the results will be noisy.
-You will find more info about the `GPS` node and functions in [Reference Manual](http://www.cyberbotics.com/reference/).
+You will find more info about the `GPS` node and functions in [Reference Manual](../reference/index.md).
```
Comment (omichel) [inline guide/programming.md]: I would say "... in **the** Reference manual".

File: guide/webots-faq.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 This chapter is a selection of frequently asked questions found on the [Webots forum](http://www.cyberbotics.com/forum).
 You may find additional information directly in the group.
-Other useful sources of information about Webots include: [Webots Reference Manual](http://www.cyberbotics.com/reference/) and [Cyberbotics' Robot Curriculum](http://en.wikibooks.org/wiki/Cyberbotics'_Robot_Curriculum).
+Other useful sources of information about Webots include: [Webots Reference Manual](../reference/index.md) and [Cyberbotics' Robot Curriculum](http://en.wikibooks.org/wiki/Cyberbotics'_Robot_Curriculum).
```
Comment (omichel) [inline guide/webots-faq.md]: I would say "... include the Webots Reference Manual and the Cyberbotics' Robot Curriculum".

**Comment by fabienrohrer** [comment]
👍

**Comment by fabienrohrer** [comment]
👍

**Pull request merged**